quarkchain/evm/messages.py

```diff
@@ -137,6 +135,23 @@ def validate_transaction(state, tx):
     if not tx.sender:  # sender is set and validated on Transaction initialization
         raise UnsignedTransaction(tx)
 
+    if tx.version == 2:
+        if (
+            state.qkc_config.ENABLE_EIP155_SIGNER_TIMESTAMP is not None
+            and state.timestamp < state.qkc_config.ENABLE_EIP155_SIGNER_TIMESTAMP
+        ):
+            raise InvalidTransaction("EIP155 Signer is not enable yet.")
+        if tx.from_chain_id != tx.to_chain_id or tx.from_shard_id != tx.to_shard_id:
```

Comment:
I guess this check is not sufficient.  What if a malicious miner replays the tx to another shard although from_chain_id == to_chain_id == malicious_chain_id != expected_chain_id?